media_manager/movies/service.py

```diff
@@ -113,6 +114,48 @@ def delete_movie_request(self, movie_request_id: MovieRequestId) -> None:
         """
         self.movie_repository.delete_movie_request(movie_request_id=movie_request_id)
 
+    def delete_movie(
+        self,
+        movie_id: MovieId,
+        delete_files_on_disk: bool = False,
+        delete_torrents: bool = False,
+    ) -> None:
+        """
+        Delete a movie from the database, optionally deleting files and torrents.
+
+        :param movie_id: The ID of the movie to delete.
+        :param delete_files_on_disk: Whether to delete the movie's files from disk.
+        :param delete_torrents: Whether to delete associated torrents from the torrent client.
+        """
+        if delete_files_on_disk or delete_torrents:
+            movie = self.movie_repository.get_movie_by_id(movie_id=movie_id)
+
+            log.debug(f"Deleting ID: {movie.id} - Name: {movie.name}")
+
+            if delete_files_on_disk and movie.library:
```

Comment:
why check if a movies library isn't none?
```suggestion
            if delete_files_on_disk:
```